aspnetcore/blazor/performance/profiling.md

```diff
@@ -0,0 +1,226 @@
+
+```html
+<script src="_framework/blazor.webassembly#[.{fingerprint}].js" 
+    autostart="false"></script>
+<script>
+  Blazor.start({
+    configureRuntime: function (builder) {
+      builder.withConfig({
+        browserProfilerOptions: {
+          sampleIntervalMs: 10,
+        }
+      });
+    }
+  });
+</script>
+```
+
+## Call specification (:::no-loc text="callspec":::)
+
+If you want to filter profiled methods, you can use call specification (:::no-loc text="callspec":::). For more information, see [Trace MonoVM profiler events during startup](https://github.com/dotnet/runtime/blob/main/docs/design/mono/diagnostics-tracing.md#trace-monovm-profiler-events-during-startup).
+
+Add `callspec` to the `browser` WebAssembly profiler in the `<WasmProfilers>` element. In the following example, the `{APP NAMESPACE}` placeholder is the app's namespace:
+
+```xml
+<WasmProfilers>browser:callspec=N:{APP NAMESPACE};</WasmProfilers>
+```
+
+Configure `callSpec` in `browserProfilerOptions`. Replace the `{APP NAMESPACE}` placeholder in the following example with the app's namespace:
+
+```html
+<script src="_framework/blazor.webassembly#[.{fingerprint}].js" 
+    autostart="false"></script>
+<script>
+  Blazor.start({
+    configureRuntime: function (builder) {
+      builder.withConfig({
+        browserProfilerOptions: {
+          callSpec: "N:{APP NAMESPACE}",
+        }
+      });
+    }
+  });
+</script>
+```
+
+## Log profiling for memory troubleshooting
+
+Enable integration with the log profiler using the `<WasmProfilers>` and `<WasmBuildNative>` properties in the app's project file (`.csproj`):
+
+```xml
+<PropertyGroup>
+  <WasmProfilers>log</WasmProfilers>
+  <WasmBuildNative>true</WasmBuildNative>
+</PropertyGroup>
+```
+
+To trigger a heap shot, add the following, where the `{APP NAMESPACE}` placeholder is the app's namespace:
+
+```csharp
+namespace {APP NAMESPACE};
+
+public class Profiling
+{
+    [JSExport]
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    public static void TakeHeapshot() { }
+}
+```
+
+Invoke `TakeHeapshot` to create a memory heap shot and flush the contents of the profile to the file system. Download the resulting `.mpld` file to analyze the data.
+
+## EventPipe profiler
+
+[EventPipe](/dotnet/core/diagnostics/eventpipe) is a runtime component used to collect tracing data, similar to [ETW](/windows/win32/etw/event-tracing-portal) and [perf_events](https://wikipedia.org/wiki/Perf_%28Linux%29).
+
+* Manual testing
+  * Browser developer tools: Download the `.json` output file, open the file in Visual Studio, and find the expected method calls.
+  * [`dotnet-trace`](/dotnet/core/diagnostics/dotnet-trace): Open the `.nettrace` output file in Visual Studio and find the expected method calls.
+* Web-based testing
+  * Use the JavaScript API to obtain a [promise](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Promise) of NetTrace (`.nettrace`) bytes.
```

Comment:
Promise here is technical detail that most users would not need to know. Please drop mention of it.